Plugin/src/main/java/dev/rosewood/rosestacker/manager/StackManager.java

```diff
@@ -70,6 +71,30 @@ public void reload() {
             long interval = autosaveFrequency * 20 * 60;
             this.autosaveTask = Bukkit.getScheduler().runTaskTimer(this.rosePlugin, () -> this.saveAllData(false), interval, interval);
         }
+
+        this.stackingBasedOnPerformance = ConfigurationManager.Setting.PERFORMANCE_TPS_TOGGLE.getBoolean();
+        if(this.stackingBasedOnPerformance){
+            Bukkit.getServer().getScheduler().runTaskAsynchronously(rosePlugin, new Runnable() {
+                @Override
+                public void run() {
+                    if(ConfigurationManager.Setting.PERFORMANCE_TPS_TOGGLE.getBoolean()){
+                        Bukkit.getScheduler().runTaskTimerAsynchronously(rosePlugin, () -> {
+                            double[] tps = Bukkit.getServer().getTPS();
```

Comment:
Server#getTPS() is a Paper-API method and this will throw an error on Spigot servers. I believe it's only possible to check the server TPS using NMS on Spigot servers, so that functionality will need to be added to all the NMSHandler interface and implemented for each handler.